--- pngconf.h.orig	2007-09-08 05:22:56.000000000 +0200
+++ pngconf.h	2007-09-09 00:00:00.000000000 +0200
@@ -1375,6 +1375,14 @@
 #      if 0 /* ... other platforms, with other meanings */
 #      endif
 #   endif
+
+#   if !defined(PNG_IMPEXP)
+#       include <qconfig.h>
+#       if defined(QT_VISIBILITY_AVAILABLE)
+#           define PNG_IMPEXP __attribute__((visibility("default")))
+#       endif
+#   endif
+
 #endif
 
 #ifndef PNGAPI
